dj/api/nodes.py

```diff
@@ -567,6 +590,51 @@ def create_a_source(
     return node  # type: ignore
 
 
+@router.post("/nodes/namespace/{namespace}/", status_code=201)
+def create_a_node_namespace(
+    namespace: str,
+    session: Session = Depends(get_session),
+) -> JSONResponse:
+    """
+    Create a node namespace
+    """
+    if get_node_namespace(
```

Comment:
Does this part care if you create a namespace where its parent namespace doesn't exist yet? For example can you create `basic.users` without having already created `basic`? And if so, should we just auto-create `basic` in that case as an available namespace?

Reply:
Right now this line will raise a 404 and return a json error if the namespace doesn't exist. We definitely could make it auto-create the namespace but doesn't that bring us back to the original design? I thought we were moving in the other direction to make namespaces more explicit. Some people have even asked if namespace creation can be locked down to a curator/server-admin. 😨 

I'm not convinced which direction is better. Is there any value in keeping it explicit for now and then adding the auto namespace creation later? It could also be something that's instrumented client side and implemented differently by different clients, i.e. maybe the python client automatically creates the namespace when it gets the 404 error but a UI requires an explicit namespace creation by the user.

Reply:
Yeah, that's true.. I guess I view the value of explicit namespaces as largely around being able to track the namespaces in a separate field, which lets us filter by namespace more easily, rather than trying to parse each node's name to extract the implicit namespace. That makes queries like "find all nodes under X namespace" faster to execute. It also opens things up later on, like if we wanted to add ACLs to namespaces or limit access in some other way. But I view creating us auto-creating some namespaces as acceptable, or at least something that could vary depending on the DJ deploy.

Reply:
This is a tough call. We want to make it seem like working with DJ is easy, but also we should be aware of the problem of metric proliferation. I think a middle ground could be that by default:
- the top level namespaces would require a direct call
- and inside them we could make the namespace creation automatic, with the ability to lock it down by the top-level namespace owner. 

This could offer some notion of a free but controlled environment.